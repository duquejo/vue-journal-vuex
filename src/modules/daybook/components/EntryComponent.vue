<template>
  <div 
    class="entry-container pointer p-3"
    @click="$router.push({ name: 'entry', params: { id: entry.id }})">
    <!-- Title -->
    <div class="entry-title d-flex align-items-center">
      <span class="text-success fs-5 fw-bold">{{ day }}</span>
      <span class="mx-1 fs-5">{{ month }}</span>
      <span class="mx-2 fw-light">{{ yearDay }}</span>
    </div>
    <div class="entry-description">{{ entry.text }}</div>
  </div>
</template>

<script>
const months = ['January', 'February', 'March', 'April', 'May', 'June','July', 'August', 'September', 'October', 'November', 'December'];
const days   = ['Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday'];

export default {
  props: {
    entry: {
      type: Object,
      required: true
    }
  },
  computed: {
    shortenText() {
      return this.entry.text.length > 130 ? 
        this.entry.text.substring(0, 130) + '...' : 
        this.entry.text;
    },
    day() {
      const date = new Date(this.entry.date);
      return date.getDate();
    },
    month() {
      const date = new Date(this.entry.date);
      return months[date.getMonth()];
    },
    yearDay() {
      const date = new Date(this.entry.date);
      return `${ date.getFullYear() }, ${ days[ date.getDay() ]}`; // Mappings
    }
  }
}
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #c7c8c9;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>