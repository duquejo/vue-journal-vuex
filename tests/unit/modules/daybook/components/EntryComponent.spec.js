import { shallowMount } from '@vue/test-utils';
import EntryComponent from '@/modules/daybook/components/EntryComponent';
import { journalState } from '../../../mocks/test-journal-state';

describe('EntryComponent.vue', () => {

  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(EntryComponent, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
      mocks: {
        $router: mockRouter,
      }
    }
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should redirect when entry-container is clicked', () => {
    wrapper.find('.entry-container').trigger('click');
    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: journalState.entries[0].id
      }
    });
  });

  it('should have the computed variables populated', () => {
    expect(wrapper.vm.day).toBe(9);
    expect(wrapper.vm.month).toBe('September');
    expect(wrapper.vm.yearDay ).toBe('2023, Saturday');
  });
});