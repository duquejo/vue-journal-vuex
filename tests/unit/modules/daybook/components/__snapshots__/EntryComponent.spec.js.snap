// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryComponent.vue should match the snapshot 1`] = `
<div class="entry-container pointer p-3">
  <!-- Title -->
  <div class="entry-title d-flex align-items-center"><span class="text-success fs-5 fw-bold">9</span><span class="mx-1 fs-5">September</span><span class="mx-2 fw-light">2023, Saturday</span></div>
  <div class="entry-description">asdasdasd</div>
</div>
`;
